lib/money/money.rb

```diff
@@ -343,29 +343,26 @@ def split(num)
     calculate_splits(num).sum([]) { |value, count| Array.new(count, value) }
   end
 
-  # Calculate the splits evenly without losing pennies.
-  # Returns the number of high and low splits and the value of the high and low splits.
-  # Where high represents the Money value with the extra penny
-  # and low a Money without the extra penny.
+  # Sum splits allows to perform arithmetic operations on the splits
+  # and get the resulting Money. This method should be used when it's not
+  # a requirement to get the splits. This will skip the Array creation for
+  # better performance.
   #
-  # @param [2] number of parties.
+  # @param [2] number of parties
   #
-  # @return [Hash<Money, Integer>]
+  # @return Money
   #
   # @example
-  #   Money.new(100, "USD").calculate_splits(3) #=> {Money.new(34) => 1, Money.new(33) => 2}
-  def calculate_splits(num)
-    raise ArgumentError, "need at least one party" if num < 1
-    subunits = self.subunits
-    low = Money.from_subunits(subunits / num, currency)
-    high = Money.from_subunits(low.subunits + 1, currency)
-
-    num_high = subunits % num
-
-    {}.tap do |result|
-      result[high] = num_high if num_high > 0
-      result[low] = num - num_high
+  #   Money.new(100, "USD").sum_splits(3) #=> Money.new(100, "USD")
+  #   Money.new(100, "USD").sum_splits(3) do |split|
+  #     split.allocate([0.9, 0.1]).last
+  #   end #=> Money.new(9.99, "USD")
+  def sum_splits(num)
+    unless block_given?
```

Comment:
Should we have an arg check on `num` here to?

```rb
raise ArgumentError, 'need at least one party' if num < 1
```

Reply:
It's going to get picked up by `calculate_split` anyway. Not sure how much more we get out of doing the check again.

Reply:
Yeah I always struggle with these kinds of guards, so I'd love extra opinions here; the part that tipped me in favour of adding it at this level is this being a public method with the other being private. 

I think it's nicer when public methods are very specific about what they expect?